init/Kconfig

```diff
@@ -1927,14 +1927,27 @@ config HAS_CARGO
 config HAS_CARGO_XBUILD
   def_bool $(success,cargo xbuild --version)
 
-config RUST
+config MENU_RUST
   bool "Enables building kernel modules written in Rust"
   depends on HAS_RUST
   depends on HAS_CARGO
   depends on HAS_CARGO_XBUILD
   help
     Whether to support building modules written in Rust.
 
+config RUST_DISABLE
+        bool
+        depends on RUST_MENU
+        help
+          This option disables the support for Rust, so that make 
+          allyesconfig and make allmodconfig will not enable it. To 
+          build modules written in Rust, leave this option set to 'n'.
+
+config RUST
+        bool
+        default y
+        depends on RUST_MENU && !RUST_DISABLE
```

Comment:
I don't think any of this is needed -- just leave it "config RUST" with a depends on HAS_RUST that does the "cargo xbuild" test. An "allmodconfig" will not include things that require RUST if HAS_RUST isn't set, etc.

Reply:
Are you saying the whole `RUST_DISABLE` should be removed entirely, or just that this particular line can be simplified?